File: android/app/src/main/kotlin/com/frag/noteflutter/Database/UserModel.kt
package com.frag.noteflutter.Database

import androidx.room.Entity
import androidx.room.PrimaryKey


@Entity
data class UserModel(
    @PrimaryKey(autoGenerate = true) var id : Int = 0,
    var title : String,
    var description : String,
    var date : String
)

File: lib/Database/UserModel.dart
class UserModel{
  late int id;
  late String title;
  late String description;
  late String date;

  UserModel(int id , String title , String description , String date){
    this.id = id;
    this.title = title;
    this.description = description;
    this.date = date;
  }


  Map<String, dynamic> toJson() => {
    'id' : id,
    'title': title,
    'description': description,
    'date': date,
  };

  UserModel.fromJson(Map<String , dynamic> json) :
        id = json["id"] ,
        title = json["title"] ,
        description = json["description"] ,
        date = json["date"];
}